[processing] Fix "Extract by expression" algorithm short help string

agiudiceandrea opened the pull request:
## Description

The "Extract by expression" (native:extractbyexpression) QGIS native C++ algorithm short help string contains an invalid link to `{qgisdocs}/user_manual/working_with_vector/expression.html`: the `{qgisdocs}` placeholder is only available for QGIS Python algorithms help description.

In order to fix the issue, I replaced the string
`For more information about expressions see the <a href =\"qgisdocs}/user_manual/working_with_vector/expression.html\">user manual</a>`
with the string
`For help with QGIS expression functions, see the inbuilt help for specific functions which is available in the expression builder.`
which is also used in the "Geometry by expression" algorithm.

This PR needs to be backported.

<!--
  BEFORE HITTING SUBMIT -- Please BUILD AND TEST your changes thoroughly. This is YOUR responsibility! Do NOT rely on the QGIS code maintainers to do this for you!!

  IMPORTANT NOTES FOR FIRST TIME CONTRIBUTORS
  ===========================================

  Congratulations, you are about to make a pull request to QGIS! To make this as easy and pleasurable for everyone, please take the time to read these lines before opening the pull request.

  Include a few sentences describing the overall goals for this pull request (PR). If applicable also add screenshots or - even better - screencasts.
  Include both: *what* you changed and *why* you changed it.

  If this is a pull request that adds new functionality which needs documentation, give an especially detailed explanation.
  In this case, start with a short abstract and then write some text that can be copied 1:1 to the documentation in the best case.

  Also mention if you think this PR needs to be backported. And list relevant or fixed issues.

------------------------

  Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by checking the following list.
  Feel free to ask in a comment if you have troubles with any of them.

  - Commit messages are descriptive and explain the rationale for changes.

  - Commits which fix bugs include `Fixes #11111` at the bottom of the commit message. If this is your first pull request and you forgot to do this, write the same statement into this text field with the pull request description.

  - New unit tests have been added for relevant changes

  - You have run the `scripts/prepare_commit.sh` script (https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit.
    If you didn't do this, you can also run `./scripts/astyle_all.sh` from your source folder.

  - You have read the QGIS Coding Standards (https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
-->

agiudiceandrea committed:
[processing] Fix "Extract by expression" help string

removing the not working link to the online documentation

nyalldawson commented:
@DelazJ are you happy with this change?

3nids commented:
I would either go for `#ifdef SIP_RUN` to keep it available for Python docs or make that `{qgisdocs}` is also available in Doxygen.

DelazJ commented:
Yes, we don't necessarily want people browse the docs here but find the functions to use, and this change addresses it better imo. We will still keep the link in the alg description in docs.
Note that the link in "Extract by expression" does not work but afaict the one in "Select by expression" does (probably because it's in yaml?). Align the description?

agiudiceandrea committed:
[processing] Fix "Select by expression" help string

removing the link to the user manual online

agiudiceandrea committed:
[processing] Fix "Order by expression" help string

adding a reference the the inbuilt help for specific functions which is available in the expression builder

agiudiceandrea commented:
> Note that the link in "Extract by expression" does not work but afaict the one in "Select by expression" does (probably because it's in yaml?). Align the description?

Yes, the {qgisdocs} placeholder is only available for QGIS Python algs.
It is replaced in https://github.com/qgis/QGIS/blob/2d1aa68f0d044f2aced7ebeca8d2fa6b754ac970/python/plugins/processing/algs/help/__init__.py#L60

I've aligned the description of "Select by expression" and "Order by expression", too.

agiudiceandrea commented:
Should it be backported?

DelazJ commented:
I think there's at least a bugfix here so good to have in next ltr.
